src/query/block/types.go

```diff
@@ -167,6 +182,7 @@ func (m Metadata) String() string {
 type Builder interface {
 	AppendValue(idx int, value float64) error
 	AppendValues(idx int, values []float64) error
+	SetBlockType(blockType BlockType)
```

Comment:
Instead of using `SetBlockType` can we just force it to be supplied in the constructor to the builder?

Seems like people could forget otherwise rather than forcing them to supply in the constructor.

Reply:
Discussed offline re: adding a `BuildAsType` method to `Builder` instead to make this more explicit as most usages of `Build` want default typing